File: client/src/app/_models/hr/candidateAssessment.ts
import { ICandidateAssessmentItem } from "./candidateAssessmentItem";

export interface ICandidateAssessment
{
     id: number;
     orderItemId: number;
     candidateId: number;
     
     customerName: string;
     categoryRefAndName: string;
     orderDate: Date;

     assessedByEmployeeName: string;
     assessedOn: Date;
     assessResult: string;
     requireInternalReview: boolean;
     checklistHRId: number;
     taskIdDocControllerAdmin: number;
     remarks: string;
     candidateAssessmentItems: ICandidateAssessmentItem[];
}
export class CandidateAssessment implements ICandidateAssessment
{
   
     id= 0;
     orderItemId =0;

     customerName='';
     categoryRefAndName = '';
     orderDate = new Date();
     
     candidateId= 0;
     assessedByEmployeeName= '';
     assessedOn= new Date();
     assessResult= "";
     requireInternalReview = false;
     checklistHRId= 0;
     remarks= '';
     cvRefId= 0;
     taskIdDocControllerAdmin= 0;
     candidateAssessmentItems: ICandidateAssessmentItem[]=[];
}

File: api/Entities/HR/CandidateAssessment.cs
using System.ComponentModel.DataAnnotations;
using api.Entities.Admin;
using api.Entities.Admin.Order;

namespace api.Entities.HR
{
    public class CandidateAssessment: BaseEntity
    {
        [Required]
        public int OrderItemId { get; set; }
        public int CandidateId {get; set;}
        public string CustomerName { get; set; }
        public string CategoryRefAndName { get; set; }
        public DateTime AssessedOn { get; set; }
        public int AssessedByEmployeeId { get; set; }
        public string AssessedByEmployeeName {get; set;}
        [Required]
        public bool RequireInternalReview {get; set;}
        public int ChecklistHRId {get; set;}
        [Required]
        public string AssessResult { get; set; } = "Not Assessed";
        public string Remarks { get; set; }
        public int CVRefId { get; set; }        //this field is set once this candidate is referred to client - CVRef
        public int TaskIdDocControllerAdmin {get; set;}
        public OrderItem OrderItem { get; set; }
        public ICollection<CandidateAssessmentItem> CandidateAssessmentItems { get; set; }
    }
}